### PR TITLE
Add backup notice

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/10021

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Clean profile
- Enable Rewards
- Restore wallet with user funds
- Wait for `OnFetch` callback
- Click on gear to bring up backup/restore modal
- "Backup" tab should be selected by default
- Verify that modal text matches spec
- Verify that clicking the "connect to Uphold" link initiates the Uphold connection process
- Verify that you can switch back and forth between Backup/Restore tabs
-----
- Clean profile
- Enable Rewards
- Claim promotion
- Wait for `OnFetch` callback
- Click on gear to bring up backup/restore modal
- "Restore" tab should be selected by default
- Should be unable to switch to "Backup" tab; instead, clicking "Backup" will display a modal notice
- Verify that clicking the "connect to Uphold" link in the modal notice initiates the Uphold connection process

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
@emerick can we please move text for new users into backup tab, so that we are not stacking dialogs.

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc [hunk 1]
```diff
@@ -299,7 +299,7 @@ void Create::RegisterPersonaCallback(
   ledger_->SetAutoContributeEnabled(true);
   braveledger_state::SetPaymentId(ledger_, payment_id);
   braveledger_state::SetAnonymousCardId(ledger_, card_id);
-  braveledger_state::SetFetchOldBalanceEnabled(ledger_, false);
+  //braveledger_state::SetFetchOldBalanceEnabled(ledger_, false);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc:302]: @NejcZdovc I wasn't quite sure how to deal with this one. It seems that we reset this flag to false after registering the persona, which interferes with these browser tests.

**Comment by emerick** [comment]
@NejcZdovc Updated with browser tests but there's one small outstanding issue, see my comment above.

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc]: what we can do is wrap this line in `!ledger::is_testing`

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc]: Perfect, will do.

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/create.cc]: Fixed.

**Comment by emerick** [comment]
CI had a browser test failure on linux, restarting there. Ignoring test-install failure on Mac.

**Comment by emerick** [comment]
All CI is green now except for previously-noted test-install failure on Mac.

**Pull request merged**